### ISSUE TITLE
Fatal signal 11 (SIGSEGV), code 1, fault addr 0x2f0015 in tid 18380, Face Recognizer

### ISSUE THREAD

**Issue by nikoskoutr** [Itseez/opencv#6524]
### Please state the information for your system
- OpenCV version: 3.1.0
- Host OS: Linux (Arch) 
- Target OS: Android 6.0
- Compiler & CMake: arm-linux-androideabi-g++ (GCC) 4.9.x 20150123 5.3 & cmake version 3.5.2

### In which part of the OpenCV library you got the issue?
Face Recognition

### Expected behaviour
Creating and training a face recognizer

### Actual behaviour
When i try to train the face recognizer i create, i get a fatal error:
Fatal signal 11 (SIGSEGV), code 1, fault addr 0x2f0015 in tid 18380, Face Recognizer

### Additional description
I use javacv to create an android application for face recognition. I have tried rebuilding javacv from source and i am still getting the same error. I opened an issue on javacv and after some searching they thought that this might be an opencv issue. Here is the link to the issue i opened:
https://github.com/bytedeco/javacv/issues/400

Update: After rebuilding javacv from source i get the error:
Fatal signal 11 (SIGSEGV), code 1, fault addr 0x8 in tid 7174

Everything else i use from opencv (mat, normalize function etc) works fine, the only time i get the error is when i try to train the facerecognizer. When i provide less images than it expects, i get the error that the facerecognizer needs more images to work, so it begins to work and brakes somewhere inside the code.

### Code example to reproduce the issue / Steps to reproduce the issue
The whole project is in my github that can be build with android studio and run on an android device.
```
FaceRecognizer model = opencv_face.createFisherFaceRecognizer();
model.train(images, labels);
```

**Comment by alalek** [comment]
1) "face" is contrib module, so issue should go [there](https://github.com/Itseez/opencv_contrib/issues)
2) Provided information is not very helpful. Please add more info, like stack traces and exceptions output.
3) We don't maintain javacv, so some reproducer without it will be useful.
4) There is no "official" OpenCV Java bindings for this module (possible due some problems)

**Comment by nikoskoutr** [comment]
Sorry for the lack of information, as i searched better for how to get a more detailed tombstone i have found a way. 
Here is the detailed crash dump:
[http://pastebin.com/x5Rzu2vq](url)

I can see that the error i get if from MatVector put function. Is it used by the face module?
I posted again here just because i saw that the MatVector was in the crash dump, if you still think that the issue is from the face module i will refer to the appropriate section.

**Comment by mshabunin** [comment]
`MatVector` is JavaCV class.

We will close the issue for now, feel free to reopen it if more details arrive.

I'd recommend you to try to reproduce and debug it using desktop java application, maybe with debug libraries if available.
